packages/react-components/src/queries/useLatestAssetPropertyValues/useLatestAssetPropertyValues.ts

```diff
@@ -0,0 +1,184 @@
+   * happen in the same task and the resulting refetchInterval
+   * executes within the batch timeout window.
+   */
+  const syncFlag = useSyncLatestAssetPropertyValueQueries({
+    enabled,
+    refreshRate,
+  });
+
+  const { getAssetPropertyValue, batchGetAssetPropertyValue } =
+    useIoTSiteWiseClient({ iotSiteWiseClient });
+
+  const queries = useMemo(() => {
+    return requests.map((request) => {
+      return {
+        enabled:
+          syncFlag &&
+          enabled &&
+          clientIsValid({
+            getAssetPropertyValue,
+            batchGetAssetPropertyValue,
+          }) &&
+          requestIsValid(request),
+        queryKey: new LatestAssetPropertyValueKeyFactory({
+          ...request,
+          refreshRate,
+        }).create(),
+        queryFn: createLatestAssetPropertyValueQueryFn(
+          {
+            getAssetPropertyValue,
+            batchGetAssetPropertyValue,
+          },
+          {
+            batchSechedulerTimeout,
+            batchSize,
+          }
+        ),
+        refetchInterval: refreshRate,
+      };
+    });
+  }, [
+    getAssetPropertyValue,
+    batchGetAssetPropertyValue,
+    enabled,
+    syncFlag,
+    requests,
+    refreshRate,
+    batchSechedulerTimeout,
+    batchSize,
+  ]);
+
+  return useQueries(
+    {
+      queries,
+    },
+    queryClient
+  );
+};
+
+export const createLatestAssetPropertyValueQueryFn = (
+  client?: QueryFnClient,
+  settings?: Pick<
+    UseLatestAssetPropertyValuesOptions,
+    'batchSechedulerTimeout' | 'batchSize'
+  >
+) => {
+  return async ({
+    queryKey: [{ assetId, propertyId, propertyAlias }],
+    signal,
+  }: QueryFunctionContext<
+    ReturnType<LatestAssetPropertyValueKeyFactory['create']>
+  >) => {
+    const { getAssetPropertyValue, batchGetAssetPropertyValue } = client ?? {};
+
+    invariant(
+      clientIsValid(client),
+      'Expected getAssetPropertyValue or batchGetAssetPropertyValue to be defined as required by the enabled flag.'
+    );
+
+    invariant(
+      requestIsValid({ assetId, propertyId, propertyAlias }),
+      'Expected assetId and propertyId to be defined or propertyAlias to be defined as required by the enabled flag.'
+    );
+
+    if (batchGetAssetPropertyValue) {
+      return await LatestAssetPropertyValueBatcher.getInstance({
+        batchGetAssetPropertyValue,
+        ...settings,
+      }).send(
+        {
+          assetId,
+          propertyId,
+          propertyAlias,
+        },
+        {
+          abortSignal: signal,
+        }
+      );
+    }
+
+    if (getAssetPropertyValue) {
+      return await new GetLatestAssetPropertyValueRequest({
+        getAssetPropertyValue,
+      }).send(
+        {
+          assetId,
+          propertyId,
+          propertyAlias,
+        },
+        {
+          abortSignal: signal,
+        }
+      );
+    }
+
+    throw 'Expected request to be handled by one of the two client functions as required by the enabled flag.';
```

Comment:
Shouldn't this be handled by the invariant check for the client?

Reply:
Yes, it is handled by the invariant, but because it's an either or check they're both technically undefined. This was the easiest way to ensure that the function doesn't optionally return undefined since it's impossible for it to do so.